infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -148,21 +147,14 @@ public long apply(LogReplicationEntryMsg msg) throws ReplicationWriterException
 
         // Skip entries that have already been processed
         if (msg.getMetadata().getTimestamp() <= lastMsgTs) {
-            log.warn("Ignore Log Entry. Received message with snapshot {} is smaller than lastMsgTs {}.",
-                    msg.getMetadata().getSnapshotTimestamp(), lastMsgTs);
+            log.warn("Ignore Log Entry. Received message with timestamp {} is smaller than lastMsgTs {}.",
+                    msg.getMetadata().getTimestamp(), lastMsgTs);
             return Address.NON_ADDRESS;
-        }
-
-        //If the entry is the expecting entry, process it and process
-        //the messages in the queue.
-        if (msg.getMetadata().getPreviousTimestamp() == lastMsgTs) {
```

Comment:
same here.. can you explain why this check was removed?

Reply:
Same explanation as above